processor/src/chiplets/memory/mod.rs

```diff
@@ -168,7 +170,7 @@ impl Memory {
 
                     // compute delta as difference between context IDs, addresses, or clock cycles
                     let delta = if prev_ctx != ctx {
-                        (ctx - prev_ctx) as u64
+                        (ctx - prev_ctx).into()
```

Comment:
I would first convert the contexts into integers and then perform the subtraction as logically, we should not be able to subtract one context from another.

Reply:
Done [36e09d1](https://github.com/0xPolygonMiden/miden-vm/pull/1117/commits/36e09d13068176f944fba2907d57557016dc5518)

Although I'd argue that it *needs* to make sense to "take the difference of 2 context IDs", well, because we're doing it here. By converting to `u32` and then subtracting in a way makes it less clear what the type, since with that hack I can also add context IDs, divide them, etc.

Or in other words, if it actually doesn't make sense to subtract context IDs, then is there a way to redesign this code snippet to not rely on it? Could the `delta` here be computed only from the addresses or the clock cycles?

Reply:
Thinking about this some more, it already makes sense to compare `ContextId`s (we derive `PartialOrd` and `Ord`); then why not subtract them? Basically a `ContextId` differs from, say, an `AccountId` in that the root context is the smallest, and every `call()` creates a "greater context ID". The difference between 2 `ContextId`s would be formally defined as the number of `call`s from the root context; which is how we think of the type, and how we use it here.

If you disagree, let's just leave it as is and not spend too much time on this 🙂

Reply:
> The difference between 2 `ContextId`s would be formally defined as the number of `call`s from the root context

Similar but not exactly like you described. Difference between two contexts is actually the number of clock cycles between these contexts were created. So, root context is created at cycle 0, the next context could be created at cycle 123 and then the difference between them would be 123. The next context could be created as cycles 155, and then the difference between them would be 32. But 32 itself is not a valid context (for execution of this program).

Basically, what we want to do is `ContextId - ContextId = ClockCycleDelta` but I think this probably introduces more complexity than we need. So, let's keep it as is :)

Reply:
Ah, that makes a lot of sense. Agreed!